packages/financial-templates-lib/src/price-feed/ForexDailyPriceFeed.js

```diff
@@ -0,0 +1,271 @@
+          )}`
+        );
+        console.log(
+          `- ⚠️  If you want to manually verify the specific exchange prices, you can make a GET request to: \n- https://exchangeratesapi.io/history?base=${this.base}&symbols=${this.symbol}`
+        );
+        console.groupEnd();
+      }
+      return returnPrice;
+    }
+
+    returnPrice = match.closePrice;
+    if (verbose) {
+      console.group(`\n(${this.symbol}${this.base}) Historical Daily Price @ ${match.closeTime}`);
+      console.log(`- ✅ Close Price:${this.web3.utils.fromWei(returnPrice.toString())}`);
+      console.log(
+        `- ⚠️  If you want to manually verify the specific exchange prices, you can make a GET request to: \n- https://exchangeratesapi.io/history?base=${this.base}&symbols=${this.symbol}`
+      );
+      console.groupEnd();
+    }
+    return returnPrice;
+  }
+
+  getHistoricalPricePeriods() {
+    return this.historicalPricePeriods.map(historicalPrice => {
+      return [historicalPrice.closeTime, historicalPrice.closePrice];
+    });
+  }
+
+  getLastUpdateTime() {
+    return this.lastUpdateTime;
+  }
+
+  getLookback() {
+    return this.lookback;
+  }
+
+  getPriceFeedDecimals() {
+    return this.priceFeedDecimals;
+  }
+
+  async update() {
+    const currentTime = this.getTime();
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== undefined && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "ForexDailyPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTimes + this.minTimeBetweenUpdates - currentTime
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "ForexDailyPriceFeed",
+      message: "Updating ForexDailyPriceFeed",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime
+    });
+
+    // Find the closest day that completed before the beginning of the lookback window, and use
+    // it as the start date.
+    const startLookbackWindow = currentTime - this.lookback;
+    const startDateString = this._secondToDateTime(startLookbackWindow);
+    const endDateString = this._secondToDateTime(currentTime);
+
+    // 1. Construct URL.
+    // See https://exchangeratesapi.io/ for how this url is constructed.
+    const url = [
+      "https://api.exchangeratesapi.io/history?",
+      `start_at=${startDateString}&end_at=${endDateString}`,
+      `&base=${this.base}&symbols=${this.symbol}`
+    ].join("");
+
+    // 2. Send request.
+    const historyResponse = await this.networker.getJson(url);
+
+    // 3. Check responses.
+    if (
+      !historyResponse ||
+      !historyResponse.rates ||
+      Object.keys(historyResponse.rates).length === 0 ||
+      !historyResponse.rates[Object.keys(historyResponse.rates)[0]][this.symbol]
+    ) {
```

Comment:
Thoughts on doing this to check every element and simplify a bit?

```suggestion
    if (
      !historyResponse?.rates
      Object.keys(historyResponse.rates).length === 0 ||
      Object.values(historyResponse.rates).some(rate => !rate[this.symbol])
    ) {
```